src/client/Microsoft.Identity.Client/Internal/JsonWebToken.cs

```diff
@@ -181,9 +181,15 @@ internal class JWTPayload
             public string Issuer { get; set; }
 
             [JsonProperty(JsonWebTokenConstants.ReservedClaims.NotBefore)]
+#if SUPPORTS_SYSTEM_TEXT_JSON
+            [JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
+#endif
             public long ValidFrom { get; set; }
 
             [JsonProperty(JsonWebTokenConstants.ReservedClaims.ExpiresOn)]
+#if SUPPORTS_SYSTEM_TEXT_JSON
+            [JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
```

Comment:
Is this needed? This class is used to go from OBJ to JSON, not the other way around?

Reply:
Counter-point: I would it add it here for the sake of consistency - add the attribute for any integer properties.